### DISCUSSION TITLE
Can an R-based script be submmited using kbatch?

### DISCUSSION THREAD

**Discussion by zhb1234zhb** [microsoft/PlanetaryComputer#250] [Q&A]
Hi,
I've been using PlanetaryComputer for some time now and have found them extremely useful. However, I found there could be a time limit for the notebook server. My scripts need relatively long time to search and process Landsat images, so I need to run them in batch mode. In the online documents, I also found that there is kbatch for python, but is there any similar way to run R scripts?
Thanks.

Hongbo

**Comment by gilbertocamara** [comment]
Dear @zhb1234zhb The R package "sits" provides an end-to-end environment for processing Landsat and Sentinel images in Planetary Computer. The package is available on CRAN and documentation is in https://e-sensing.github.io/sitsbook/

**Reply by zhb1234zhb** [reply]
Dear Gilberto, Thank you for sharing the information with me. The package and book both appear to be of great interest. I was wondering if you could clarify one thing for me, if I utilize the functions from "sits" to establish a connection to the Planetary Computer, will my R script be subjected to the same time limit as that of the notebook server, i.e. 24 hours? Thank you very much for your kind assistance.

**Reply by gilbertocamara** [reply]
Dear @zhb1234zhb All external packages such as `sits` are subject to the same limits. The difference is that `sits` provides support for building a regular data cube to which one can do time series analysis using deep learning.

**Reply by zhb1234zhb** [reply]
Thanks a lot for your explanations. I believe your package will be very helpful for my future work, though currently, my major concern is the time limit.

**Comment by zhb1234zhb** [comment]
Update:
I installed kbatch in the R environment and tried to submit my R code using the following command:
kbatch job submit --name=j1 --image=mcr.microsoft.com/planetary-computer/r --args='["R", "CMD BATCH test.R"]' --code="test.R"
However, when I typed the command "kbatch job list -o table", no job is printed.

**Comment by TomAugspurger** [comment]
> I also found that there is kbatch for python, but is there any similar way to run R scripts?

`kbatch` is just a way for submitting jobs to run on a Kubernetes cluster. It's implemented in Python, and has a Python-based CLI, but it doesn't (or shouldn't) have any requirements at runtime.

That said, you *might* be better of exploring how to run jobs on your own compute. I haven't had a ton of time to work on `kbatch` so it has plenty of rough edges. As long as the compute is in Azure, you'll still have the same fast access to the data and APIs as you get on the Planetary Computer Hub.
